test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,417 @@
+
+    It "Verify Protect-CmsMessage emits recipient errors" {
+        try {
+            "Hello World" | Protect-CmsMessage -To "SomeThumbprintThatDoesNotExist" -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.ProtectCmsMessageCommand"
+        }
+    }
+
+    It "Verify CmsMessage cmdlets works with paths" {
+        try {
+            $randomNum = Get-Random -Minimum 1000 -Maximum 9999
+            $tempPath = Join-Path $TestDrive "$randomNum-Path-Test-File"
+            $encryptedPath = $tempPath + ".encrypted.txt"
+            "Hello World","How are you?" | Set-Content $tempPath
+
+            Protect-CmsMessage -Path $tempPath -To $certLocation -OutFile $encryptedPath
+
+            $message = Get-CmsMessage -LiteralPath $encryptedPath
+            $message.Recipients.Count | Should Be 1
+            $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+            $expected = "Hello World" + [System.Environment]::NewLine + "How are you?" + [System.Environment]::NewLine
+            $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+            $decrypted | Should Be $expected
+
+            $decrypted = Unprotect-CmsMessage -Path $encryptedPath -To $certLocation
+            $decrypted | Should Be $expected
+        } finally {
+            Remove-Item $tempPath, $encryptedPath -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    It "Verify Unprotect-CmsMessage works with local store" {
+        try {
+            $randomNum = Get-Random -Minimum 1000 -Maximum 9999
+            $tempPath = Join-Path $TestDrive "$randomNum-Path-Test-File"
+            "Hello World" | Protect-CmsMessage -To $certLocation -OutFile $tempPath
+
+            # Decrypt using $importedCert in the Cert store
+            $decrypted = Unprotect-CmsMessage -Path $tempPath
+            $decrypted | Should Be "Hello World"
+        } finally {
+            Remove-Item $tempPath -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    It "Verify Unprotect-CmsMessage emits recipient errors" {
+        try {
+            "" | Unprotect-CmsMessage -To "SomeThumbprintThatDoesNotExist" -IncludeContext -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.UnprotectCmsMessageCommand"
+        }
+    }
+
+    It "Verify failure to extract Ascii armor generates an error" {
+        try {
+            "Hello World" | Unprotect-CmsMessage -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "InputContainedNoEncryptedContentIncludeContext,Microsoft.PowerShell.Commands.UnprotectCmsMessageCommand"
+        }
+
+        # Should have round-tripped content
+        $result = "Hello World" | Unprotect-CmsMessage -IncludeContext
```

Comment:
This looks like it should be a separate It. The errors are similar, but the tests are different.